misk/src/main/kotlin/misk/digester/MergingDigest.kt

```diff
@@ -0,0 +1,351 @@
+package misk.digester
+
+import misk.proto.tdigest.Centroid
+import misk.proto.tdigest.MergingDigestData
+import kotlin.math.PI
+import kotlin.math.asin
+
+class MergingDigest {
+
+    //amount of compression to data
+    var compression: Double = 0.0
+
+    //main list of centroids
+    var mainCentroids: MutableList<Centroid> = mutableListOf()
+    //total weight of unmerged main centroids
+    var mainWeight: Double = 0.0
+
+    //centroids that have been added but not yet merged into main list
+    var tempCentroids: MutableList<Centroid> = mutableListOf()
+    //total weight of unmerged temp centroids
+    var tempWeight: Double = 0.0
+
+    var min: Double = Double.POSITIVE_INFINITY
+    var max: Double = Double.NEGATIVE_INFINITY
+
+    constructor()
+
+    constructor(compression: Double) {
+      this.compression = compression
+    }
+
+    constructor (
```

Comment:
This shouldn't be a public constructor. Callers should not be able to set internals.